editor/store/store.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import refx from 'refx';
+import multi from 'redux-multi';
+import { get, flowRight } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { dispatch, getState, subscribe } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { mobileMiddleware } from '../utils/mobile';
+import effects from './effects';
+
+/**
+ * This function applies the custom middlewares used specifically in the editor moodule
+ * It also restricts the getState call to the module's partial state only.
+ *
+ * @return {Object} Redux custom store
+ */
+function applyMiddlewaresAndRestrictState() {
+	const middlewares = [
+		mobileMiddleware,
+		refx( effects ),
+		multi,
+	];
+
+	const enhancedGetState = () => get( getState(), 'core/editor' );
```

Comment:
Instead of restricting by overriding `getState`, do you think in this model of data management we might want to create our own variant of `connect` which accepts the sub-tree to select from?

```js
connect( 'core/editor' )( mapStateToProps )( MyComponent )
```

Reply:
Yeah, no real preference here on my side, I do think your option is more flexible especially when we'll start to query other modules' state. This was just a convenience for now to avoid changing all the components.